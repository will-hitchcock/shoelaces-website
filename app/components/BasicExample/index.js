/**
*
* BasicExample
*
*/

import React from 'react'
import { Container, Row, Col } from 'react-shoelaces'
import Box from 'components/Box'
import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live'

const scope = { Container, Row, Col, Box }

const code =
`<Container>
    <Row>
      <Col xs={1 / 5}><Box /></Col>
      <Col xs={1 / 5}><Box /></Col>
      <Col xs={1 / 5}><Box /></Col>
      <Col xs={1 / 5}><Box /></Col>
      <Col xs={1 / 5}><Box /></Col>
    </Row>
    <Row>
      <Col xs={1 / 3}><Box /></Col>
      <Col xs={1 / 3}><Box /></Col>
      <Col xs={1 / 3}><Box /></Col>
    </Row>
    <Row>
      <Col xs={1 / 2}><Box /></Col>
      <Col xs={1 / 2}><Box /></Col>
    </Row>
  </Container>
`
const BasicExample = () =>
  <LiveProvider scope={scope} code={code}>
    <LivePreview />
    <Col xs={8 / 12} offset={2 / 12}>
      <LiveEditor />
      <LiveError />
    </Col>
  </LiveProvider>

export default BasicExample
