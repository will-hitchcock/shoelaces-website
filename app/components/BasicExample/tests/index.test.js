import React from 'react'
import BasicExample from '../index'

describe('<BasicExample />', () => {
  it('renders a <BasicExample />', () => {
    expect(shallow(<BasicExample />)).toMatchSnapshot()
  })
})
