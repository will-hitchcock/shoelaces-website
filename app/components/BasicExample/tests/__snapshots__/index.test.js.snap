exports[`<BasicExample /> renders a <BasicExample /> 1`] = `
<LiveProvider
  code="<Container>
    <Row>
      <Col xs={1 / 5}><Box /></Col>
      <Col xs={1 / 5}><Box /></Col>
      <Col xs={1 / 5}><Box /></Col>
      <Col xs={1 / 5}><Box /></Col>
      <Col xs={1 / 5}><Box /></Col>
    </Row>
    <Row>
      <Col xs={1 / 3}><Box /></Col>
      <Col xs={1 / 3}><Box /></Col>
      <Col xs={1 / 3}><Box /></Col>
    </Row>
    <Row>
      <Col xs={1 / 2}><Box /></Col>
      <Col xs={1 / 2}><Box /></Col>
    </Row>
  </Container>
"
  mountStylesheet={true}
  noInline={false}
  scope={
    Object {
      "Box": [Function],
      "Col": [Function],
      "Container": [Function],
      "Row": [Function],
    }
  }>
  <LivePreview />
  <styled.div
    offset={0.16666666666666666}
    xs={0.6666666666666666}>
    <LiveEditor />
    <LiveError />
  </styled.div>
</LiveProvider>
`;
